src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -46,6 +46,7 @@ PreReleaseRequest<I>::~PreReleaseRequest() {
   if (!m_shutting_down) {
     m_image_ctx.state->handle_prepare_lock_complete();
   }
+  delete m_on_finish;
```

Comment:
Right. There is a scenario of double free, as `m_on_finish->complete();` is called before the destructor is executed. And the `complete()` deletes the object pointed to by `m_on_finish`.

@badone I think in this particular case, we have to ignore what coverity scan reports, as m_on_finish is deleted indirectly.